counters.hh

```diff
@@ -18,45 +18,17 @@
 class mutation;
 class atomic_cell_or_collection;
 
-class counter_id {
-    int64_t _least_significant;
-    int64_t _most_significant;
+class counter_id : public utils::tagged_uuid<counter_id> {
```

Comment:
Why not: using counter_id = utils::tagged_uuid<class something>? Avoids a wrapper class.

Reply:
For `generate_random` below, but this is only a technical step for the patchset.
It is replaced in the next patch with `using ...`